src/main/cpp/VideoState.cpp

```diff
@@ -1542,3 +1619,375 @@ void VideoState::sdl_audio_callback(Uint8 *stream, int len) {
 	}
 }
 
+void VideoState::set_speed(const int step) {
+	int i;
+	for (i = 0; i < FF_ARRAY_ELEMS(rate_speed_map); i++) {
+		if (last_speed == rate_speed_map[i].clock_speed) {
+			if (i+step >= 0 && i+step < FF_ARRAY_ELEMS(rate_speed_map)) {
+				if (!isPaused() && rate_speed_map[i + step].clock_speed == 0) {
+					toggle_pause();
+					return; // We can't process the 0 as a new speed for the presentation time
+				}
+				if (!muted && rate_speed_map[i + step].clock_speed != 1) {
+					toggle_mute();
+				}
+				else if (muted && rate_speed_map[i + step].clock_speed == 1) {
+					toggle_mute();
+				}
+#if CONFIG_AVFILTER
+				vfilters = rate_speed_map[i + step].command;
+#else
+				pts_speed = rate_speed_map[i + step].pts_speed;
+#endif
+				last_speed = rate_speed_map[i + step].clock_speed;
+				newSpeed_req = 1;
+				continue_read_thread.notify_one();
+				return;
+			}
+		}
+	}
+}
+
+int VideoState::get_master_clock_speed() {
+	int speed = 0;
+	if (this->video_stream >= 0
+		&& this->audio_stream >= 0) {
+		switch (this->get_master_sync_type()) {
+		case AV_SYNC_VIDEO_MASTER:
+			speed = pVidclk->get_clock_speed();
+			break;
+		case AV_SYNC_AUDIO_MASTER:
+			speed = pAudclk->get_clock_speed();
+			break;
+		default:
+			speed = pExtclk->get_clock_speed();
+			break;
+		}
+	}
+	return speed;
+}
+
+#if CONFIG_AVFILTER
+// From cmdutil class
+double VideoState::get_rotation(AVStream *st) {
```

Comment:
I'm not sure if we ever need this. It's to rotate the image in the video...

But I'm fine with leaving it here for now.

Reply:
We don't really need it for our case